test/run_tests.py

```diff
@@ -96,68 +112,110 @@ def http_server():
         serverlog.close()
 
 
-def execute_tests(interpreter, browser, cli_options):
+def execute_tests(interpreter, browser, suite, sauceusername, saucekey):
     runner = [interpreter, '-m', 'robot.run']
     options = [opt.format(browser=browser,
                           py_version=interpreter + sys.version[:3])
                for opt in ROBOT_OPTIONS]
-    if env.TRAVIS:
-        options.extend(get_travis_conf(browser))
-    command = runner + options + cli_options + [env.ACCEPTANCE_TEST_DIR]
-    log_start(command)
+    if sauceusername and saucekey:
+        options.extend(get_sauce_conf(browser, sauceusername, saucekey))
+    command = runner
+    if suite:
+        command += ['-s', suite]
+    command += options + [env.ACCEPTANCE_TEST_DIR]
+    log_start(command, sauceusername, saucekey)
     syslog = os.path.join(env.RESULTS_DIR, 'syslog.txt')
-    # Running tests as an external process, not using `robot_cli`, to allow
-    # using different interpreter that is used for running this script.
-    subprocess.call(command, env=dict(os.environ, ROBOT_SYSLOG_FILE=syslog))
-
-
-def get_travis_conf(browser):
-    conf = [
-        '--variable', 'SAUCE_USERNAME:{}'.format(env.SAUCE_USERNAME),
-        '--variable', 'SAUCE_ACCESS_KEY:{}'.format(env.SAUCE_ACCESS_KEY)
-    ]
-    if browser == 'firefox':
-        conf.extend([
-            '--variable',
-            'REMOTE_URL:http://{}:{}@ondemand.saucelabs.com:80/wd/hub'.format(
-                env.SAUCE_USERNAME, env.SAUCE_ACCESS_KEY
-            )
-        ])
-    else:
-        conf.extend([
-            '--variable',
-            'DESIRED_CAPABILITIES:build:{0}-{1},tunnel-identifier:{0}'.format(
-                env.TRAVIS_JOB_NUMBER, browser
-            )
-        ])
-    return conf
+    subprocess.call(
+        command, env=dict(os.environ, ROBOT_SYSLOG_FILE=syslog)
+    )
 
 
-def log_start(command_list):
+def log_start(command_list, sauceusername, saucekey):
```

Comment:
Could just have the signature as `command_list, *hidden` and then later `for item in hidden`.